core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -751,6 +739,18 @@ abstract Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
         Number valueForSearch(Number value) {
             return value;
         }
+
+        /**
+         * Returns true if the object is a number and has a decimal part
+         */
+        protected boolean hasDecimalPart(Object number) {
+            if (number instanceof Number) {
+                double doubleValue = ((Number) number).doubleValue();
+                return doubleValue % 1 != 0;
+            }
+            return false;
```

Comment:
If `number` is a string like `1.1` then the parsing would fail with `java.lang.NumberFormatException: For input string: "1.1"`. This is already the case in the existing `parse` methods. So I think it should be fine to return `false` here.